/*
 * Copyright 2010-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
 * that can be found in the license/LICENSE.txt file.
 */

package org.jetbrains.kotlin.idea.filters;

import com.intellij.testFramework.TestDataPath;
import org.jetbrains.kotlin.test.JUnit3RunnerWithInners;
import org.jetbrains.kotlin.test.KotlinTestUtils;
import org.jetbrains.kotlin.test.TargetBackend;
import org.jetbrains.kotlin.test.TestMetadata;
import org.junit.runner.RunWith;

import java.io.File;
import java.util.regex.Pattern;

/** This class is generated by {@link org.jetbrains.kotlin.generators.tests.TestsPackage}. DO NOT MODIFY MANUALLY */
@SuppressWarnings("all")
@TestMetadata("idea/testData/debugger/exceptionFilter")
@TestDataPath("$PROJECT_ROOT")
@RunWith(JUnit3RunnerWithInners.class)
public class KotlinExceptionFilterTestGenerated extends AbstractKotlinExceptionFilterTest {
    public void testAllFilesPresentInExceptionFilter() throws Exception {
        KotlinTestUtils.assertAllTestsPresentByMetadata(this.getClass(), new File("idea/testData/debugger/exceptionFilter"), Pattern.compile("^([^\\.]+)$"), TargetBackend.ANY, false);
    }

    @TestMetadata("breakpointReachedAt")
    public void testBreakpointReachedAt() throws Exception {
        KotlinTestUtils.runTest("idea/testData/debugger/exceptionFilter/breakpointReachedAt/", this::doTest);
    }

    @TestMetadata("inlineFunCallInLibrary")
    public void testInlineFunCallInLibrary() throws Exception {
        KotlinTestUtils.runTest("idea/testData/debugger/exceptionFilter/inlineFunCallInLibrary/", this::doTest);
    }

    @TestMetadata("inlineFunFromLibrary")
    public void testInlineFunFromLibrary() throws Exception {
        KotlinTestUtils.runTest("idea/testData/debugger/exceptionFilter/inlineFunFromLibrary/", this::doTest);
    }

    @TestMetadata("inlineFunInnerClassFromLibrary")
    public void testInlineFunInnerClassFromLibrary() throws Exception {
        KotlinTestUtils.runTest("idea/testData/debugger/exceptionFilter/inlineFunInnerClassFromLibrary/", this::doTest);
    }

    @TestMetadata("inlineFunctionAnotherFile")
    public void testInlineFunctionAnotherFile() throws Exception {
        KotlinTestUtils.runTest("idea/testData/debugger/exceptionFilter/inlineFunctionAnotherFile/", this::doTest);
    }

    @TestMetadata("inlineFunctionAnotherFileWithSmapApplied")
    public void testInlineFunctionAnotherFileWithSmapApplied() throws Exception {
        KotlinTestUtils.runTest("idea/testData/debugger/exceptionFilter/inlineFunctionAnotherFileWithSmapApplied/", this::doTest);
    }

    @TestMetadata("inlineFunctionInnerClass")
    public void testInlineFunctionInnerClass() throws Exception {
        KotlinTestUtils.runTest("idea/testData/debugger/exceptionFilter/inlineFunctionInnerClass/", this::doTest);
    }

    @TestMetadata("inlineFunctionObjectLiteral")
    public void testInlineFunctionObjectLiteral() throws Exception {
        KotlinTestUtils.runTest("idea/testData/debugger/exceptionFilter/inlineFunctionObjectLiteral/", this::doTest);
    }

    @TestMetadata("inlineFunctionSameFile")
    public void testInlineFunctionSameFile() throws Exception {
        KotlinTestUtils.runTest("idea/testData/debugger/exceptionFilter/inlineFunctionSameFile/", this::doTest);
    }

    @TestMetadata("kotlinClass")
    public void testKotlinClass() throws Exception {
        KotlinTestUtils.runTest("idea/testData/debugger/exceptionFilter/kotlinClass/", this::doTest);
    }

    @TestMetadata("kt2489")
    public void testKt2489() throws Exception {
        KotlinTestUtils.runTest("idea/testData/debugger/exceptionFilter/kt2489/", this::doTest);
    }

    @TestMetadata("kt2489_2")
    public void testKt2489_2() throws Exception {
        KotlinTestUtils.runTest("idea/testData/debugger/exceptionFilter/kt2489_2/", this::doTest);
    }

    @TestMetadata("librarySources1")
    public void testLibrarySources1() throws Exception {
        KotlinTestUtils.runTest("idea/testData/debugger/exceptionFilter/librarySources1/", this::doTest);
    }

    @TestMetadata("librarySources2")
    public void testLibrarySources2() throws Exception {
        KotlinTestUtils.runTest("idea/testData/debugger/exceptionFilter/librarySources2/", this::doTest);
    }

    @TestMetadata("librarySources3")
    public void testLibrarySources3() throws Exception {
        KotlinTestUtils.runTest("idea/testData/debugger/exceptionFilter/librarySources3/", this::doTest);
    }

    @TestMetadata("librarySources4")
    public void testLibrarySources4() throws Exception {
        KotlinTestUtils.runTest("idea/testData/debugger/exceptionFilter/librarySources4/", this::doTest);
    }

    @TestMetadata("multiSamePackage")
    public void testMultiSamePackage() throws Exception {
        KotlinTestUtils.runTest("idea/testData/debugger/exceptionFilter/multiSamePackage/", this::doTest);
    }

    @TestMetadata("simple")
    public void testSimple() throws Exception {
        KotlinTestUtils.runTest("idea/testData/debugger/exceptionFilter/simple/", this::doTest);
    }
}
